CMakeLists.txt

```diff
@@ -321,6 +321,10 @@ if(SWIFT_PROFDATA_FILE AND EXISTS ${SWIFT_PROFDATA_FILE})
   add_definitions("-fprofile-instr-use=${SWIFT_PROFDATA_FILE}")
 endif()
 
+set(SWIFT_TOOLS_INSTALL_DIR "bin" CACHE PATH
```

Comment:
Should we ensure that this matches the install bin dir?

Reply:
I tried to search for such a parameter I could reuse, but I was not able to find it -- it seems we hardcode `bin` and/or we rely on preexisting defaults (like it was the case for `llvm_install_symlink` pre LLVM 15)

Updating all the places to use a proper parameter would require more work and qualification outside of the scope of unblocking `next` -- so in this context I leaned toward adding the variable to the main `CMakeLists.txt` to "highlight" this concern in future reviews.

Reply:
After a bit more investigation, wondering if I should instead solve this issue inside of [tools/driver/CMakeLists.txt](https://github.com/apple/swift/blob/dc8d37d6b5adeb2371755844445ef5b3021c789a/tools/driver/CMakeLists.txt#L114) -- in particular noticing that in there `add_swift_tool_symlink` and `swift_install_in_component` are laying down the same symlinks during installation (albeit with different means), so there could be a chance to simplify this part instead of adding to it

https://github.com/apple/swift/blob/dc8d37d6b5adeb2371755844445ef5b3021c789a/tools/driver/CMakeLists.txt#L114

https://github.com/apple/swift/blob/dc8d37d6b5adeb2371755844445ef5b3021c789a/tools/driver/CMakeLists.txt#L123-L125

https://github.com/apple/swift/pull/6053 seems to suggest that may have been the original intent for adding `add_swift_tool_symlink` in the first place

Reply:
Depending on how we want to do it, we can base this on `${LLVM_TOOLS_INSTALL_DIR}` or `${CMAKE_INSTALL_BINDIR}`.

Reply:
Now I see -- I though the question was about the risk of installing the symlinks in a different directory that the target program, but instead it seems we were more concerned on using a sane default value for the setting.

I reverted to the previous approach and used `${CMAKE_INSTALL_BINDIR}` as the default value in e7c4b19a8ec